extensions/circuit/src/adapters/alu.rs

```diff
@@ -219,8 +219,10 @@ where
             let mut c_bytes = [0u8; READ_BYTES];
             c_bytes[0] = c_u32 as u8;
             c_bytes[1] = (c_u32 >> 8) as u8;
-            c_bytes[2] = (c_u32 >> 16) as u8;
-            c_bytes[3] = (c_u32 >> 16) as u8;
+            let bit_extension = (c_u32 >> 16) as u8;
+            for byte in &mut c_bytes[2..] {
+                *byte = bit_extension;
+            }
```

Comment:
This adapter was used in both 32 and 64 bits. The old code had a bug that negative values wouldn't be properly extended to 64 bits.